src/plugins/run/annotations.ts

```diff
@@ -0,0 +1,203 @@
+import { Dictionary } from '@stoplight/types';
+import { parse } from '@stoplight/yaml';
+import * as unified from 'unified';
+
+import { MDAST } from '../../ast-types';
+
+// When we have bandwidth, might make sense to look into the new lower level pattern made possible by micromark ecosystem
+// Example in the gfm plugin - https://github.com/remarkjs/remark-gfm/blob/main/index.js#L30-L32
+export const smdAnnotations: unified.Attacher = function () {
+  return function transform($root) {
+    const root = $root as MDAST.Root;
+    const nodes = root.children;
+
+    const processed: MDAST.Content[] = [];
+
+    let inTab: boolean = false;
+    let skipNext: boolean = false;
+
+    // temporary variable for storing current tabContainer
+    let tabPlaceholder: MDAST.Tabs = {
+      type: 'tabs',
+      data: {
+        hName: 'tabs',
+      },
+      children: [
+        {
+          type: 'tab',
+          data: {
+            hName: 'tab',
+          },
+          children: [],
+        },
+      ],
+    };
+
+    for (const i in nodes) {
+      if (!nodes[i]) continue;
+
+      if (skipNext) {
+        skipNext = false;
+        continue;
+      }
+
+      // this node
+      const node = nodes[i];
+
+      // next node
+      const next = nodes[+i + 1] ? nodes[+i + 1] : null;
```

Comment:
it'd let us do
```suggestion
      const next = nodes[i + 1] ? nodes[i + 1] : null;
```
though I have to admit I'm not a fan of OOB access, but let it be.

Reply:
What's the benefit? If I remember correctly some tests failed when I did what you're suggesting, but I can double check.

Reply:
What was breaking? I think you applied my suggestion before `entries()` and that's why.
Sorry if I wasn't clear.